mcm/tests/api_tools.py

```diff
@@ -0,0 +1,496 @@
+            headers=full_headers,
+        )
+
+    def lucene_request(
+        self, endpoint: str, method: str, data: dict = {}, headers: dict = {}
+    ) -> Response:
+        """
+        Sends a HTTP request to CouchDB Lucene.
+
+        Args:
+            endpoint (str): Endpoint to consume, e.g: entity/document?p=1
+            method (str): HTTP method, e.g: POST
+            data (dict | None): Data to include in the request body.
+            headers (dict): Request headers
+        """
+        return self.__http_request(
+            base=self.mcm_couchdb_lucene_url,
+            endpoint=endpoint,
+            method=method,
+            data=data,
+            headers=headers,
+        )
+
+    def check_test_users(self) -> bool:
+        """
+        Checks that there is one user per role
+        available in the database.
+
+        Returns:
+            True if there is one user per role,
+                False otherwise.
+        """
+        all_roles = [r.value for r in Roles]
+        query = {"selector": {"_id": {"$in": all_roles}}}
+        response = self.couchdb_request(
+            endpoint="users/_find", method="POST", data=query
+        )
+        content = response.json()
+        docs = content.get("docs", [])
+        return len(docs) == len(all_roles)
+
+    def __include_test_users(self):
+        """
+        Includes one test user per available role.
+        """
+        new_user_mock = {
+            "notes": "I am test user :)",
+            "seen_notifications": [],
+            "pwg": [],
+            "email": "example@example.com",
+            "history": [],
+        }
+        new_users: list[dict] = []
+        all_roles = [r.value for r in Roles]
+
+        for role in all_roles:
+            new_user = deepcopy(new_user_mock)
+            new_user["_id"] = role
+            new_user["username"] = role
+            new_user["role"] = role
+            new_user["fullname"] = f"Test user for role: {role}"
+            new_users += [new_user]
+
+        # Send the request.
+        self.couchdb_request(
+            endpoint="users/_bulk_docs", method="POST", data={"docs": new_users}
+        )
+
+    def mcm_client(self, role: Roles) -> McM:
+        """
+        Return an instance of the McM client to perform requests
+        to the web application.
+
+        Args:
+            role (Roles): Role to impersonate in the requests.
+        """
+        return McM(api=self, role=role)
+
+
+class McM:
+    """
+    A simple version for McM REST client adapted to
+    be used for building the test cases.
+
+    Arguments:
+        api (APIRequest): Client to perform HTTP requests.
+        role (Roles): User role to impersonate when performing the
+            requests.
+    """
+
+    def __init__(self, api: APIRequest, role: Roles):
+        self.api: APIRequest = api
+        self.role: Roles = role
+        self.logger = self.__logger()
+
+    def __logger(self) -> logging.Logger:
```

Comment:
This function is only used in the constructor, merge it.

Reply:
This is only related to style. I just want to instantiate elements in the constructor and leave its configuration to other methods. It is easier to read